pkg/api/v1beta1/dynakube/dynakube_conditions.go

```diff
@@ -0,0 +1,99 @@
+package dynakube
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// TokenConditionType identifies the token validity condition.
+	TokenConditionType string = "Tokens"
+
+	// APITokenConditionType identifies the API Token validity condition.
+	APITokenConditionType string = "APIToken"
+
+	// PaaSTokenConditionType identifies the PaaS Token validity condition.
+	PaaSTokenConditionType string = "PaaSToken"
+
+	// DataIngestTokenConditionType identifies the DataIngest Token validity condition.
+	DataIngestTokenConditionType string = "DataIngestToken"
+)
+
+// Possible reasons for ApiToken and PaaSToken conditions.
+const (
+	// ReasonTokenReady is set when a token has passed verifications.
+	ReasonTokenReady string = "TokenReady"
+
+	// ReasonTokenError is set when an unknown error has been found when verifying the token.
+	ReasonTokenError string = "TokenError"
+
+	ReasonCreated         string = "Created"
+	ReasonError           string = "Error"
+	ReasonUnexpectedError string = "UnexpectedError"
+	ReasonUpToDate        string = "UpToDate"
+)
+
+// ActiveGate related conditions.
+const (
+	ActiveGateConnectionInfoConditionType string = "ActiveGateConnectionInfo"
+	ActiveGateStatefulSetConditionType    string = "ActiveGateStatefulSet"
+	ActiveGateVersionConditionType        string = "ActiveGateVersion"
+)
+
+func (dynakube *DynaKube) SetActiveGateConnectionInfoCondition(err error) error {
+	if err != nil {
+		meta.SetStatusCondition(&dynakube.Status.Conditions, metav1.Condition{
+			Type:    ActiveGateConnectionInfoConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  ReasonUnexpectedError,
+			Message: err.Error(),
+		})
+
+		return err
+	}
+
+	meta.SetStatusCondition(&dynakube.Status.Conditions, metav1.Condition{
+		Type:   ActiveGateConnectionInfoConditionType,
+		Status: metav1.ConditionTrue,
+		Reason: ReasonCreated,
+	})
+
+	return err
+}
+
+func (dynakube *DynaKube) SetActiveGateStatefulSetErrorCondition(err error) error {
+	if err != nil {
+		meta.SetStatusCondition(&dynakube.Status.Conditions, metav1.Condition{
+			Type:    ActiveGateStatefulSetConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  ReasonError,
+			Message: err.Error(),
+		})
+
+		return err
+	}
+
+	return err
+}
+
+func (dynakube *DynaKube) SetActiveGateVersionCondition(err error) error {
+	if err != nil {
+		meta.SetStatusCondition(&dynakube.Status.Conditions, metav1.Condition{
+			Type:    ActiveGateVersionConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  ReasonError,
+			Message: err.Error(),
+		})
+
+		return err
+	}
+
+	meta.SetStatusCondition(&dynakube.Status.Conditions, metav1.Condition{
+		Type:    ActiveGateVersionConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  ReasonUpToDate,
+		Message: dynakube.Status.ActiveGate.VersionStatus.ImageID,
+	})
```

Comment:
Pretty much a continuation of my previous comment.

There is no reason for bloating the dynakube package with logic that is only relevant in a sub package.

A package-specific condition logic should be defined in that package, [example](https://github.com/Dynatrace/dynatrace-operator/blob/37afc6130005c92ba0ff4fddb503292be48269aa/pkg/controllers/dynakube/connectioninfo/oneagent/conditions.go#L1-L23)

Reasoning: Why would you want to modify an ActiveGateStatefulSet condition in, for example, a OneAgent related package? There is no good reason for that (just bad design maybe 😅 ).
Each package/reconciler/controller should care about only their conditions, not anyone else's.

It only makes sense to store more general/common conditions and condition-related helpers in 1 package, and we already have that. (see my previous comment)